### PR TITLE
Breakup, splashing, and wall film models

### PR DESCRIPTION
**Pull request by ldowen**
This PR contains

- KHRT breakup model
- ETAB breakup model
- Splash model detailed in "Development and application of a drop-wall interaction model at high ambient pressure conditions" from Ahamed et al. in Atomization and Sprays, Vol. 32, Iss. 4 2022
- A crude wall film model, assumes wall film is stationary and 1D
- Routines to allow particle creation during particle update process, involves data movement from host and device for GPU solutions

### CODE REVIEW

File: Exec/SprayTests/PeleC/SprayA_wbreakup/chem_files/mechanism.H [hunk 1]
```diff
@@ -0,0 +1,5063 @@
+#ifndef MECHANISM_H
```
Comment (marchdf) [inline Exec/SprayTests/PeleC/SprayA_wbreakup/chem_files/mechanism.H]: Feels like the chem files should belong in PelePhysics?

File: Exec/SprayTests/PeleC/SprayA_wbreakup/GNUmakefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# AMReX
```
Comment (marchdf) [inline Exec/SprayTests/PeleC/SprayA_wbreakup/GNUmakefile:1]: Part of me is a bit worried about all these Exec tests in PeleMP. I can easily see us changing things in LMeX or C and forgetting that these exist. Maybe we should think about moving these problems into the actual code bases?

File: Exec/SprayTests/PeleC/SprayA_wbreakup/ref/roi.dat
Comment (marchdf) [inline Exec/SprayTests/PeleC/SprayA_wbreakup/ref/roi.dat:1]: This is a large data file. Does it belong here?

**Comment by baperry2** [comment]
Another thing that would be super helpful here because we're not as familiar with the models being implemented is some brief documentation on the theory/models, how to run them, and the expected results from a test case.

File: Source/PP_Spray/SprayInjection.H [hunk 3]
```diff
@@ -204,9 +226,17 @@ SprayParticleContainer::uniformSprayInit(
   }
   part_vals[SprayComps::pstateT] = T_part;
   part_vals[SprayComps::pstateDia] = dia_part;
-  for (int spf = 0; spf < SPRAY_FUEL_NUM; ++spf) {
```
Comment (baperry2) [inline Source/PP_Spray/SprayInjection.H:207]: Was this removed intentionally? Test cases in LMeX and C are failing with NaNs due to these components not being initialized.

File: Source/PP_Spray/SpraySetup.cpp [hunk 4]
```diff
@@ -125,16 +130,59 @@ SprayParticleContainer::readSprayParams(int& particle_verbose)
...
   }
 
-  Real parcel_size = 1;
   Real spray_ref_T = 300.;
   bool splash_model = false;
+  int breakup_model = 0;
   //
   // Set the number of particles per parcel
   //
-  pp.query("parcel_size", parcel_size);
```
Comment (baperry2) [inline Source/PP_Spray/SpraySetup.cpp:134]: The test cases in LMeX and C are set up to have parcel_size of 10 and 100, so this change, which effectively makes parcel_size always 1 without splash/breakup models, leads to different results for those tests. Running the old MP code with a parcel_size of 1 results in no diffs vs. this code. Just want to make sure that this is the correct anticipated behavior after this PR, and we can simply remove the parcel_size input file entries going forward.

https://github.com/AMReX-Combustion/PeleLMeX/blob/56740f63b9ed9650ceb4cae171e6e561bc983431/Exec/RegTests/SprayTest/input.2d#L91

File: Source/PP_Spray/SprayInjection.H (hunk 3 above)
Comment (ldowen) [reply Source/PP_Spray/SprayInjection.H:207]: I believe this was mistakenly removed and should be added back in. Good catch

File: Source/PP_Spray/SpraySetup.cpp (hunk 4 above)
Comment (ldowen) [reply Source/PP_Spray/SpraySetup.cpp:134]: With these changes, there should not be a parcel_size input file entry. Instead, the parcel size is set based on how the particles are initialized. For jets, it is an input for the particular jet, and for uniform initialization the parcel size is set as an input here. https://github.com/AMReX-Combustion/PeleMP/blob/81ae98eeda7fd369cc12dcc57125a41be239aee9/Source/PP_Spray/SprayInjection.H#L207
The correct fix would be to have a problem specific input from the input file for the uniform spray cases that gets read in and passed through here.

File: Source/PP_Spray/SprayInjection.H (hunk 3 above)
Comment (baperry2) [reply Source/PP_Spray/SprayInjection.H:207]: Thanks for the quick replies. This has been done in https://github.com/AMReX-Combustion/PelePhysics/pull/452

File: Source/PP_Spray/SpraySetup.cpp (hunk 4 above)
Comment (baperry2) [reply Source/PP_Spray/SpraySetup.cpp:134]: Ok. Will leave this alone for now because the cases can still run, but will look into adding this later

**Pull request merged**